### PR TITLE
allocator: Block Allocate() and Release() until key list is initialized

### LINKED ISSUE #7598: Pod does not get ready (unable to create identity)
Some relevant cilium logs
```
2019-04-04T16:18:27.151769415Z level=info msg="New endpoint" containerID=f02072d889 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2516 ipv4= ipv6= k8sPodName=apps/front-24-8hpk5 subsys=endpoint

2019-04-04T16:19:57.155494166Z level=warning msg="Ongoing identity allocation has been cancelled" error="context canceled" key="k8s:app=front-24,k8s:io.cilium.k8s.policy.cluster=default,k8s:io.cilium.k8s.policy.serviceaccount=default,k8s:io.kubernetes.pod.namespace=apps" subsys=allocator
2019-04-04T16:19:57.155563498Z level=warning msg="Error changing endpoint identity" containerID=f02072d889 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2516 error="unable to resolve identity: identity allocation cancelled: context canceled" identityLabels="k8s:app=front-24,k8s:io.cilium.k8s.policy.cluster=default,k8s:io.cilium.k8s.policy.serviceaccount=default,k8s:io.kubernetes.pod.namespace=apps" ipv4= ipv6= k8sPodName=apps/front-24-8hpk5 subsys=endpoint
2019-04-04T16:19:57.155571441Z level=warning msg="Aborting endpoint join due client connection closed" containerID=f02072d889 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2516 error="context canceled" ipv4= ipv6= k8sPodName=apps/front-24-8hpk5 subsys=daemon
2019-04-04T16:19:57.173427171Z level=info msg="Removed endpoint" containerID=f02072d889 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=2516 ipv4= ipv6= k8sPodName=apps/front-24-8hpk5 subsys=endpoint
2019-04-04T16:19:57.176223517Z level=warning msg="Ignoring error while deleting endpoint" endpointID=2516 error="Unable to delete key 10.0.48.33:0 from /sys/fs/bpf/tc/globals/cilium_lxc: Unable to delete element from map cilium_lxc: no such file or directory" subsys=daemon
2019-04-04T16:19:57.176237231Z level=warning msg="Ignoring error while deleting endpoint" endpointID=2516 error="unable to remove endpoint from global policy map: Unable to delete element from map cilium_policy: no such file or directory" subsys=daemon
2019-04-04T16:19:57.176243007Z level=error msg="Endpoint cannot be created" containerID=f02072d88993890dacf42b8bf5d58d4358d7648f2bcd9a5a1a5cc68628095c25 error="aborting endpoint 2516 join due client connection closed" eventID=471b4fc0-56f5-11e9-8507-42010ac80130 subsys=daemon
```

I've also noticed in the kvstore there are a couple of locks taken which they don't seem to be released(?)

```
cilium/state/identities/v1/locks/k8s:app=front-24;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=default;k8s:io.kubernetes.pod.namespace=apps;/5b9569e8e9929e0e

cilium/state/identities/v1/locks/k8s:app=front-24;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=default;k8s:io.kubernetes.pod.namespace=apps;/5b9569e8e9929e59

cilium/state/identities/v1/locks/k8s:app=front-24;k8s:io.cilium.k8s.policy.cluster=default;k8s:io.cilium.k8s.policy.serviceaccount=default;k8s:io.kubernetes.pod.namespace=apps;/5b9569e8e9929ea1
```

(there are 290 lines like those but only 205 cilium-agents running)

### PR DESCRIPTION
**Pull request by tgraf**
It has been observed in #7598 that the allocation occurs in parallel to receiving the initial list of keys. This has the potential to corrupt the local cache as events and local modifications are unordered. Bock allocation and release until the initial list of keys has been received. This has no effect on the allocation of local, well-known or fixed identities as they are handled in the identity layer.

Also provide better visibility into what is going in the allocator in case this does will not resolve a particular issue such as #7598

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7658)

### CODE REVIEW

File: pkg/kvstore/allocator/allocator.go [hunk 1]
```diff
@@ -639,6 +649,14 @@ func (a *Allocator) GetByID(id idpool.ID) (AllocatorKey, error) {
 // the last user has released the ID, the key is removed in the KVstore and
 // the returned lastUse value is true.
 func (a *Allocator) Release(ctx context.Context, key AllocatorKey) (lastUse bool, err error) {
+	log.WithField(fieldKey, key).Info("Releasing key")
```
Comment (aanm) [inline pkg/kvstore/allocator/allocator.go:652]: Same here

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/kvstore/allocator/allocator.go [hunk 2]
```diff
@@ -538,7 +538,13 @@ func (a *Allocator) Allocate(ctx context.Context, key AllocatorKey) (idpool.ID,
 		k     = key.GetKey()
 	)
 
-	kvstore.Trace("Allocating key", nil, logrus.Fields{fieldKey: key})
+	log.WithField(fieldKey, key).Info("Allocating key")
```
Comment (aanm) [inline pkg/kvstore/allocator/allocator.go:541]: The same here is for this Info message to be changed to something else.

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by tgraf** [comment]
test-me-please

```
00:07:58.940  Stderr: VBoxManage: error: Could not open the medium '/root/VirtualBox VMs/test_runtime_1554888162270_70437/Snapshots/{1ab8b8d5-a106-4073-baa0-fdee0c542f0e}.vmdk'.
```

**Comment by tgraf** [comment]
test-me-please

File: pkg/kvstore/allocator/allocator.go (hunk 2 above)
Comment (tgraf) [reply pkg/kvstore/allocator/allocator.go:541]: Discussed on Slack, Info is fine

**Comment by tgraf** [comment]
@aanm Please have look and decide whether you agree or not. The CI is passing.

**Review by aanm — APPROVED** [review]

**Pull request merged**